test/hardhat/RouterUpgrade.test.ts

```diff
@@ -408,3 +564,29 @@ describe("Router Upgrade", function () {
     });
   });
 });
+
+// Returns the first instance of an event log from a transaction receipt that matches the address provided
+function extractSingleLog<T extends Interface, E extends EventFragment>(
```

Comment:
i think we have this function already in `Router.test.ts` - can maybe extract to test utils